pkg/controllers/provisioner/v1alpha1/controller.go

```diff
@@ -54,8 +59,19 @@ func (c *Controller) Reconcile(object controllers.Object) error {
 		return fmt.Errorf("Listing unscheduled pods, %w", err)
 	}
 
-	// 2. SKIP FOR NOW, Attempt to schedule pods on existing capacity
-	// 3. SKIP FOR NOW, Attempt to schedule remaining pods by preempting existing pods
+	unschedulable := []*v1.Pod{}
+	for _, pod := range pods.Items {
+		for _, condition := range pod.Status.Conditions {
+			if condition.Type == v1.PodScheduled && condition.Reason == v1.PodReasonUnschedulable {
```

Comment:
May be we can do this later, but wait for few seconds for a pod to be in pending state. 
Let's add a comment for now.

Reply:
I'm not sure what the difference is here. If the pod is unschedulable, it means the scheduler failed right? Is there a step I'm missing?